src/Transport/Sending/MessageDispatcher.cs

```diff
@@ -128,25 +129,21 @@ public async Task Dispatch(TransportOperations outgoingMessages, TransportTransa
                 }
                 // Accessing azureServiceBusTransaction.CommittableTransaction will initialize it if it isn't yet
                 // doing the access as late as possible but still on the synchronous path.
-                dispatchTasks.Add(DispatchBatchForDestination(destination, azureServiceBusTransportTransaction?.ServiceBusClient, azureServiceBusTransportTransaction?.Transaction, messagesToSend, cancellationToken));
+                dispatchTasks.Add(DispatchBatchForDestination(destination, azureServiceBusTransportTransaction?.ServiceBusClient, azureServiceBusTransportTransaction?.Transaction, messagesToSend, messagesThatCouldntBeSent, cancellationToken));
+
+                foreach (var message in messagesThatCouldntBeSent)
+                {
+                    dispatchTasks.Add(DispatchForDestination(destination, azureServiceBusTransportTransaction?.ServiceBusClient, azureServiceBusTransportTransaction?.Transaction, message, cancellationToken));
+                }
             }
         }
 
-        async Task DispatchBatchForDestination(string destination, ServiceBusClient? client, Transaction? transaction, Queue<ServiceBusMessage> messagesToSend, CancellationToken cancellationToken)
+        async Task DispatchBatchForDestination(string destination, ServiceBusClient? client, Transaction? transaction,
+            Queue<ServiceBusMessage> messagesToSend, List<ServiceBusMessage> messagesThatCouldntBeSent,
+            CancellationToken cancellationToken)
         {
-            var messageCount = messagesToSend.Count;
             int batchCount = 0;
             var sender = messageSenderRegistry.GetMessageSender(destination, client);
-            // There are two limits for batching that unfortunately are not enforced over TryAdd.
-            //
-            // Limit 1: For transactional sends you cannot add more than 100 messages into the same batch. This limit
```

Comment:
I realized Limit1 is also not needed because as part of https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/842 this global check was introduced https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/842/files#diff-c6b0be4814ff55f20eda0f3d7d1cf54000b898f72b4c5fbefbe004793f3a39b8R89 which throws before we even attempt to batch

Reply:
The global check is necessary to avoid handing over things to the broker. It would lead to zombie messages